tx-pool/src/service.rs

```diff
@@ -473,17 +493,75 @@ impl TxPoolServiceBuilder {
     }
 
     /// Start a background thread tx-pool service by taking ownership of the Builder, and returns a TxPoolController.
-    pub fn start(self, network: NetworkController) {
+    pub fn start(self, network: NetworkController) -> Result<(), AnyError> {
         let last_txs_updated_at = Arc::new(AtomicU64::new(0));
         let consensus = self.snapshot.cloned_consensus();
         let tx_pool = TxPool::new(
-            self.tx_pool_config,
+            Arc::new(self.tx_pool_config),
             self.snapshot,
             Arc::clone(&last_txs_updated_at),
         );
 
+        let pool_cache = if tx_pool.config.cache_file.exists() {
```

Comment:
suggest to extracting code to a `load_tx_pool` fn and use a background thread to add tx entry to the pool one by one.

Reply:
I will extract code, **but I don't want to use a background thread**.

**In fact, I want to block all processes (but, not yet) until the tx pool is ready from the recovery.**
It's the simplest way to avoid few issues.
For example, I think the cached transactions should have the highest priority, but if network is working parallel, some other conflict transactions may be insert before these transactions.
And this node may request already-existed transactions through network.